pytorch_blade/src/compiler/mlir/converters/impl/shapes.cpp

```diff
@@ -426,6 +426,95 @@ bool ConvertAtenFlip(MhloConversionContext& ctx, const torch::jit::Node& node) {
+  auto loc = GetNodeLocation(ctx, node);
+  auto jit_tensor = node.input(0);
+  auto jit_other = node.input(1);
+  auto ml_tensor = ctx.GetMlirValue(jit_tensor);
+  auto ml_other = ctx.GetMlirValue(jit_other);
+
+  auto& builder = *ctx.builder;
+  auto result_type = BuildMlirRankedTensorType(builder, *node.output(0));
+
+  auto other_ty = ml_other.getType().cast<mlir::RankedTensorType>();
+  auto other_shape = other_ty.getShape();
+  auto rank = other_ty.getRank();
+  llvm::SmallVector<mlir::Value> dim_values;
+  for (int64_t idx = 0; idx < rank; ++idx) {
+    mlir::Value dim_val = other_shape[idx] == mlir::ShapedType::kDynamicSize
+        ? builder.create<mlir::tensor::DimOp>(loc, ml_other, idx).getResult()
+        : builder.create<mlir::arith::ConstantIndexOp>(loc, other_shape[idx])
+              .getResult();
+    dim_values.push_back(dim_val);
+  }
+  mlir::Value new_shape =
+      builder.create<mlir::tensor::FromElementsOp>(loc, dim_values);
+  TORCH_CHECK(result_type.getRank() == rank);
+  if (rank == 0) {
+    // if reshape output is a scalar
+    auto result =
+        builder.create<mlir::mhlo::ReshapeOp>(loc, result_type, ml_tensor);
+    ctx.value_map[node.output(0)] = result.getResult();
+  } else {
+    // if reshape output is a tensor
+    auto result = builder.create<mlir::mhlo::DynamicReshapeOp>(
+        loc, result_type, ml_tensor, new_shape);
+    ctx.value_map[node.output(0)] = result.getResult();
+  }
+  return true;
+}
+
+// chunk_size = (dim_size + chunks - 1) / chunks
+// start_index = 0;
+// for (int i = 0; i < chunks; ++i) {
+//   end_index = start_index + chunk_size > dim_size ? dim_size : start_index +
+//   chunk_size builder.create<DynamicSlice>(.., start_index, end_index,
+//   strides) start_index = end_index
+// }
+bool ConvertAtenChunk(
+    MhloConversionContext& ctx,
+    const torch::jit::Node& node) {
+  auto loc = GetNodeLocation(ctx, node);
+  auto jit_input = node.input(0);
+  auto jit_chunks = node.input(1);
+  auto jit_dim = node.input(2);
+  bool is_const_dims = IsPrimConstant(*jit_chunks) && IsPrimConstant(*jit_dim);
+  if (!is_const_dims) {
+    LOG(WARNING) << "chunks and dim must be constant for aten::chunk";
+    return false;
+  }
+
+  auto& builder = *ctx.builder;
+  auto ml_input = ctx.GetMlirValue(jit_input);
+  mlir_dim_t rank = GetRankOfMlirValue(ml_input);
+  mlir_dim_t trans_chunks = CastJitConstToInt64(*jit_chunks);
+  mlir_dim_t trans_dim = CastJitConstToInt64(*jit_dim);
+  trans_dim = NormalizeDimIndex(trans_dim, rank);
+  auto dim_size = BuildStdDimSizeOfTensor(builder, loc, ml_input, trans_dim);
+  auto chunks = BuildStdConstForI64(builder, loc, trans_chunks);
+  auto one = BuildStdConstForI64(builder, loc, 1);
+  auto dim_size_p_chunks_m_1 = builder.create<mlir::arith::SubIOp>(
+      loc, builder.create<mlir::arith::AddIOp>(loc, dim_size, chunks), one);
+  auto chunked_size =
+      builder.create<mlir::arith::DivUIOp>(loc, dim_size_p_chunks_m_1, chunks);
```

Comment:
I am curious about what if using `mlir::arith::DivIOp` here.

Reply:
it should be the same if we are sure the operands are unsigned.
in std dialect, it is agreed that all the integers are signless, which mean that the users are responsible to guarantee if they are signed or unsigned.